react/src/pages/utils/components/DetailSection.tsx

```diff
@@ -1,119 +1,451 @@
+        );
+    }
     return (
-        <>
-            {titles.map((title, index) => {
-                if (index >= Math.ceil(titles.length / 2)) {
-                    return <></>;
-                } else {
-                    return <FieldDisplay title={title} value={values[index]} />;
-                }
-            })}
-        </>
+        <MuiTextField
+            className={classes.textField}
+            margin="dense"
+            label={field.title}
+            value={formatValue(field.value)}
+            required={nonNullableFields.includes(field.fieldName)}
+            disabled={field.disableEdit}
+            onChange={e => onEdit(field.fieldName, e.target.value === "" ? null : e.target.value)}
+        />
     );
 }
 
-function RightGridFieldsDisplay({ titles, values }: GridFieldDisplayProps) {
+function GridFieldsDisplay({
+    fields,
+    editMode,
+    position,
+    onEdit,
+}: {
+    fields: Field[];
+    editMode: boolean;
+    position: "left" | "right";
+    onEdit: (fieldName: string | undefined, value: any) => void;
+}) {
+    const classes = useStyles();
+    const [enums, setEnums] = useState(undefined);
+
+    useEffect(() => {
+        fetch("/api/enums").then(async response => {
+            if (response.ok) {
+                const enums = await response.json();
+                setEnums(enums);
+            } else {
+                console.error(
+                    `GET /api/enums failed with ${response.status}: ${response.statusText}`
+                );
+            }
+        });
+    }, []);
+
     return (
-        <>
-            {titles.map((title, index) => {
-                if (index < Math.ceil(titles.length / 2)) {
+        <Grid item xs={infoWidth}>
+            {fields.map((field, index) => {
+                if (position === "left" && index >= Math.ceil(fields.length / 2)) {
+                    return <></>;
+                } else if (position === "right" && index < Math.ceil(fields.length / 2)) {
                     return <></>;
                 } else {
-                    return <FieldDisplay title={title} value={values[index]} />;
+                    return (
+                        <>
+                            {editMode ? (
+                                <Fade in={editMode} mountOnEnter unmountOnExit>
+                                    <Box>
+                                        <TextField field={field} enums={enums} onEdit={onEdit} />
+                                    </Box>
+                                </Fade>
+                            ) : null}
+                            {!editMode ? (
+                                <Fade in={!editMode} mountOnEnter unmountOnExit>
+                                    <Box className={classes.box}>
+                                        <FieldDisplay title={field.title} value={field.value} />
+                                    </Box>
+                                </Fade>
+                            ) : null}
```

Comment:
Shouldn't this be one ternary instead of two? Or just if/else, possibly even as part of the existing branching in this function?